pkg/controller/pxc/users.go

```diff
@@ -226,63 +380,101 @@ func (r *ReconcilePerconaXtraDBCluster) manageMonitorUser(cr *api.PerconaXtraDBC
 	return nil
 }
 
-func (r *ReconcilePerconaXtraDBCluster) manageXtrabackupUser(cr *api.PerconaXtraDBCluster, internalSysSecretObj *corev1.Secret) error {
-	annotationName := "grant-for-1.7.0-xtrabackup-user"
-	if internalSysSecretObj.Annotations[annotationName] == "done" {
-		return nil
-	}
+func (r *ReconcilePerconaXtraDBCluster) handleClustercheckUser(cr *api.PerconaXtraDBCluster, secrets, internalSecrets *corev1.Secret, upRes *userUpdateActions) error {
+	user := &users.SysUser{
+		Name:  users.UserClustercheck,
+		Pass:  string(secrets.Data[users.UserClustercheck]),
+		Hosts: []string{"localhost"},
+	}
+
+	// grant system user privilage
+	if cr.CompareVersionWith("1.10.0") >= 0 {
+		mysqlVersion := cr.Status.PXC.Version
+		if mysqlVersion == "" {
+			var err error
+			mysqlVersion, err = r.mysqlVersion(cr, statefulset.NewNode(cr))
+			if err != nil && !errors.Is(err, versionNotReadyErr) {
+				return errors.Wrap(err, "retrieving pxc version")
+			}
+		}
 
-	pxcUser := "root"
-	pxcPass := string(internalSysSecretObj.Data["root"])
-	if _, ok := internalSysSecretObj.Data["operator"]; ok {
-		pxcUser = "operator"
-		pxcPass = string(internalSysSecretObj.Data["operator"])
+		if mysqlVersion != "" {
+			ver, err := version.NewVersion(mysqlVersion)
+			if err != nil {
+				return errors.Wrap(err, "invalid pxc version")
+			}
+
+			if !ver.LessThan(privSystemUserAddedIn) {
+				if err := r.grantSystemUserPrivilege(cr, internalSecrets, user); err != nil {
+					return errors.Wrap(err, "clustercheck user grant system privilege")
+				}
+			}
+		}
 	}
 
-	addr := cr.Name + "-pxc-unready." + cr.Namespace + ":3306"
-	hasKey, err := cr.ConfigHasKey("mysqld", "proxy_protocol_networks")
+	// update pass
+	err := r.updateUserPass(cr, secrets, internalSecrets, user)
 	if err != nil {
-		return errors.Wrap(err, "check if congfig has proxy_protocol_networks key")
-	}
-	if hasKey {
-		addr = cr.Name + "-pxc-unready." + cr.Namespace + ":33062"
+		return errors.Wrap(err, "update clustercheck users pass")
 	}
 
-	um, err := users.NewManager(addr, pxcUser, pxcPass, cr.Spec.PXC.ReadinessProbes.TimeoutSeconds)
+	//update internalSecrets
+	internalSecrets.Data[users.UserRoot] = secrets.Data[users.UserRoot]
+	err = r.client.Update(context.TODO(), internalSecrets)
```

Comment:
patch may be better than update?

Reply:
I guess it would, this was just taken from the current impl. I'll change it.